internal/allocator/allocator.go

```diff
@@ -108,6 +108,26 @@ func (a *Allocator) SetPools(pools *config.Pools) error {
 	return nil
 }
 
+// GetPoolOwner retrieves ip pool owner for the given list of ip addresses.
+func (a *Allocator) GetPoolOwner(ips []net.IP) (*config.Pool, error) {
```

Comment:
I'd rather change poolFor returning the whole record (instead of just the name) and use it (and change all the calling places to do p.Name), rather than having a wrapper that do the lookup again

Reply:
Ok, so how about delegating the compatibility check to allocator?
Outside we fetch the pool name, and we ask hte allocator if it's compatible with our service.